bundles/org.openhab.binding.homekit/src/main/java/org/openhab/binding/homekit/internal/handler/HomekitBaseServerHandler.java

```diff
@@ -0,0 +1,564 @@
+     * Uses the /accessories endpoint.
+     * Returns an empty list if there was a problem.
+     * Requires a valid secure session.
+     *
+     * @return list of accessories (may be empty)
+     * @see <a href="https://datatracker.ietf.org/doc/html/draft-ietf-homekit-http">HomeKit HTTP</a>
+     */
+    private void fetchAccessories() {
+        logger.info("Fetching accessories for BASE thing {}", thing.getUID());
+        try {
+            // byte[] json = ipTransport.get(ENDPOINT_ACCESSORIES, CONTENT_TYPE_HAP);
+            // Accessories container = GSON.fromJson(new String(json, StandardCharsets.UTF_8), Accessories.class);
+            // TODO REMOVE TEST CODE
+            Accessories container = GSON.fromJson(TODO_REMOVE_TEST_JSON, Accessories.class);
+            // Accessories result = GSON.fromJson(TODO_REMOVE_TEST_JSON, Accessories.class);
+            if (container != null && container.accessories instanceof List<Accessory> accessoryList) {
+                accessories.clear();
+                accessories.putAll(accessoryList.stream().filter(a -> Objects.nonNull(a.aid))
+                        .collect(Collectors.toMap(a -> a.aid, Function.identity())));
+            }
+            logger.info("Fetched {} accessories", accessories.size());
+            scheduler.submit(() -> accessoriesLoaded()); // notify subclass in scheduler thread
+        } catch (Exception e) {
+            logger.warn("Failed to get accessories: {}", e.getMessage());
+        }
+    }
+
+    /**
+     * Called when the thing handler has been initialized, the pairing verified, and the accessories have been loaded.
+     * Subclasses override this to perform any processing required.
+     * This method is called in the context of a scheduler thread, to avoid blocking operations.
+     */
+    protected abstract void accessoriesLoaded();
+
+    /**
+     * Extracts the accessory ID from the thing's UID property.
+     * The UID is expected to end with "-<accessoryId>".
+     *
+     * @return the accessory ID, or null if it cannot be determined
+     */
+    protected @Nullable Integer getAccessoryId() {
+        String uidProperty = thing.getProperties().get(PROPERTY_UID);
+        if (uidProperty == null) {
+            return null;
+        }
+        int accessoryIdIndex = uidProperty.lastIndexOf("-");
+        if (accessoryIdIndex < 0) {
+            return null;
+        }
+        try {
+            return Integer.parseInt(uidProperty.substring(accessoryIdIndex + 1));
+        } catch (NumberFormatException e) {
+            return null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // this is an abstract thing with no channels, so do nothing
+    }
+
+    @Override
+    public void handleRemoval() {
+        if (isChildAccessory) {
+            updateStatus(ThingStatus.REMOVED);
+        } else {
+            updateStatus(ThingStatus.REMOVING);
```

Comment:
This will cause `IllegalArgumentException`: Illegal status REMOVING. Bindings only may set UNKNOWN, ONLINE, OFFLINE or REMOVED.

Reply:
I'm excited about this and looking forward to test it. I have a Velux Active kit (KIX 300) for controlling my skylights, and it should support HomeKit.